### PR TITLE
ONNX model support and addition of tumor segmentation model

### PR DESCRIPTION
**Pull request by andreanne-lemay**
Addition of support for ONNX models in sct_deepseg as discussed [here](https://github.com/neuropoly/spinalcordtoolbox/compare/al/onnx-model?expand=1). 

Changes:
- Add support to ONNX models on model's path verification
- Add version requirement on ivadomed (version where ONNX models were implemented)
- Add t2_tumor model

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@ cryptography
 dipy
 futures
 h5py
-ivadomed
+ivadomed>=1.2.1
```
Comment (charleygros) [inline requirements.txt:7]: I wonder if we want to carefully monitor ivadomed version (ie use `==`) as it is quite critical & lot of changes at the mment

File: spinalcordtoolbox/deepseg/models.py [hunk 2]
```diff
@@ -69,7 +73,8 @@ def is_valid(path_model):
     :param path_model: str: Absolute path to folder that encloses the model files.
     """
     name_model = path_model.rstrip(os.sep).split(os.sep)[-1]
-    return os.path.exists(os.path.join(path_model, name_model + '.pt')) and \
+    return (os.path.exists(os.path.join(path_model, name_model + '.pt')) or
+           os.path.exists(os.path.join(path_model, name_model + '.onnx'))) and \
```
Comment (charleygros) [inline spinalcordtoolbox/deepseg/models.py:77]: The priority is given to onnx file in segment_volume, so I agree with your change here ;)

File: spinalcordtoolbox/deepseg/models.py [hunk 3]
```diff
@@ -30,6 +30,10 @@
         {'url': 'https://osf.io/mfxwg/download?version=5',
          'description': 'Gray matter segmentation on mouse MRI. Data from University of Queensland.',
          'default': False},
+    't2_tumor':
+        {'url': 'https://osf.io/uwe7k/download?version=1',
```
Comment (charleygros) [inline spinalcordtoolbox/deepseg/models.py:34]: I checked this file, looks all good to me.

**Comment by charleygros** [comment]
Just double checking: is this [threshold](https://github.com/neuropoly/spinalcordtoolbox/blob/master/spinalcordtoolbox/deepseg/core.py#L24) fine with you @andreanne-lemay ?

**Comment by andreanne-lemay** [comment]
> Just double checking: is this [threshold](https://github.com/neuropoly/spinalcordtoolbox/blob/master/spinalcordtoolbox/deepseg/core.py#L24) fine with you @andreanne-lemay ?

Yes, I tested a couple of tumors and 0.9 works well!

**Pull request merged**